tests/QARegressionTests/test_core/test_test_command.py

```diff
@@ -33,6 +42,7 @@ def test_test_command_with_all_options(self):
             ],
         )
         self.assertEqual(result.exit_code, 0)
+        self.assertTrue(not self.error_message.search(result.output))

```

Comment:
I think we should check for the exact errors we are expecting instead of just checking for errors in general

Reply:
I think we want the regression tests to fail if an update causes the output to change. That would signal to the developer that the test needs to be updated

Reply:
updated tests in test_validate_command and test_test_command to include assertions for checking output in the generated report. Also changed the code to use subprocess module for executing terminal commands because the previous implementation was not waiting for full execution of command.